bus/pubsub/consumer.go

```diff
@@ -0,0 +1,136 @@
+package pubsub
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/pcelvng/task/bus/info"
+	"google.golang.org/api/option"
+)
+
+type Consumer struct {
+	// Client is a Google Pub/Sub client scoped to a single project.
+	// Clients should be reused rather than being created as needed.
+	// A Client may be shared by multiple goroutines. Consumer Client
+	client *pubsub.Client
+	sub    *pubsub.Subscription
+
+	// context for clean shutdown
+	ctx  context.Context
```

Comment:
does the ctx need to be saved?

Reply:
I think so, if we send a kill signal we would want pubsub to stop cleanly. So i've use that context in the pubsub calls.